cmd/icingadb-migrate/embed/comment_query.sql

```diff
@@ -2,7 +2,9 @@ SELECT ch.commenthistory_id, UNIX_TIMESTAMP(ch.entry_time) entry_time,
   ch.entry_time_usec, ch.entry_type, ch.author_name, ch.comment_data, ch.is_persistent,
   COALESCE(UNIX_TIMESTAMP(ch.expiration_time), 0) expiration_time,
   COALESCE(UNIX_TIMESTAMP(ch.deletion_time), 0) deletion_time,
-  ch.deletion_time_usec, ch.name, o.objecttype_id, o.name1, COALESCE(o.name2, '') name2
+  ch.deletion_time_usec,
+  COALESCE(ch.name, CONCAT(o.name1, '!', COALESCE(o.name2, ''), '!', ch.commenthistory_id, '-', ch.object_id)) name,
```

Comment:
I assume you made a tradeoff for host comments/downtimes? For such that expression would yield host!!id-id – but this makes the query less complex.

Reply:
Then what about just ch.commenthistory_id (as fallback)?

Reply:
I was not certain about the cardinality between `icinga_commenthistory` to `icinga_objects` as `o.name1` and `o.name2` are part of the latter. Thus, I wanted to be sure and have included both `commenthistory_id` and `object_id`.
Furthermore, I liked having a name being similar to the other valid entries, even if it is just being yelled into the void.

Reply:
Ok, why not.